Service/VersionManager.php

```diff
@@ -123,33 +124,43 @@ public function getActiveProvider()
     /**
      * Write a new version number to the cache and storage
      *
-     * @param Version $version
+     * @param   Version $version
+     * @throws  RuntimeException
      */
     public function writeVersion(Version $version)
     {
-        $cacheItem = $this->cache->getItem('version');
-        $cacheItem->set($version);
+        try {
+            $cacheItem = $this->cache->getItem('version');
+            $cacheItem->set($version);
 
-        $this->cache->save($cacheItem);
-        $this->writer->write($version);
+            $this->cache->save($cacheItem);
+            $this->writer->write($version);
+        } catch (InvalidArgumentException $e) {
+            throw new RuntimeException('Could not write version cache item: ' . $e->getMessage());
```

Comment:
I would keep the original exception as previous exception to make debugging easier.